Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -1196,8 +1196,8 @@ internal protected void OnTargetEvent (TargetEventArgs args)
 				activeThread = args.Thread;
 
 			if (stepTimer != null && (HasExited || args.IsStopEvent)) {
-				stepTimer.Stop (true);
-				stepTimer.Dispose ();
+				stepTimer?.Stop (true);
```

Comment:
But there's a null check right above?

Reply:
Capture stepTimer in a local varaible before it's used, and then you can guarantee it's the same one being used.

Reply:
If this code is multi-threaded, atomic swaps should be used:

```csharp
var timer = Interlocked.Exchange (stepTimer, null);
if (timer != null && ..) {
    timer.Stop (true);
}

```

Reply:
thanks for the tip!